plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -125,4 +341,109 @@ private function fetch_latest_woocommerce_version() {
 
 		return null;
 	}
+
+	/**
+	 * Record the current timestamp to the throttle log.
+	 */
+	private function record_log_timestamp() {
+		$timestamps   = get_transient( self::THROTTLE_TRANSIENT );
+		$timestamps   = is_array( $timestamps ) ? $timestamps : array();
+		$timestamps[] = time();
+		set_transient( self::THROTTLE_TRANSIENT, $timestamps, self::THROTTLE_INTERVAL );
+	}
+
+	/**
+	 * Check if logging should be throttled.
+	 *
+	 * @return bool
+	 */
+	private function should_throttle_logging() {
+		$timestamps = get_transient( self::THROTTLE_TRANSIENT );
+		if ( ! is_array( $timestamps ) ) {
+			return false;
+		}
+
+		// Remove timestamps older than the throttle interval.
+		$current_time = time();
+		$timestamps   = array_filter(
+			$timestamps,
+			function ( $timestamp ) use ( $current_time ) {
+				return ( $current_time - $timestamp ) <= self::THROTTLE_INTERVAL;
+			}
+		);
+
+		// Update the transient with the filtered timestamps.
+		set_transient( self::THROTTLE_TRANSIENT, $timestamps, self::THROTTLE_INTERVAL );
+
+		// Check if the number of logs in the interval exceeds the limit.
+		return count( $timestamps ) >= self::THROTTLE_LIMIT;
+	}
+
+	/**
+	 * Sanitize the content to exclude sensitive data.
+	 *
+	 * The trace is sanitized by:
+	 *
+	 * 1. Remove the absolute path to the WooCommerce plugin directory.
+	 * 2. Remove the absolute path to the WordPress root directory.
+	 *
+	 * For example, the trace:
+	 *
+	 * /var/www/html/wp-content/plugins/woocommerce/includes/class-wc-remote-logger.php on line 123
+	 * will be sanitized to: **\/woocommerce/includes/class-wc-remote-logger.php on line 123
+	 *
+	 * @param string $message The message to sanitize.
+	 * @return string The sanitized message.
+	 */
+	private function sanitize( $message ) {
+		if ( ! is_string( $message ) ) {
+			return $message;
+		}
+
+		$wc_path = StringUtil::normalize_local_path_slashes( WC_ABSPATH );
+		$wp_path = StringUtil::normalize_local_path_slashes( ABSPATH );
+
+		$sanitized = str_replace(
+			array( $wc_path, $wp_path ),
+			array( '**/' . dirname( WC_PLUGIN_BASENAME ) . '/', '**/' ),
+			$message
+		);
+
+		return $sanitized;
+	}
+
+	/**
+	 * Sanitize the error trace to exclude sensitive data.
+	 *
+	 * @param array|string $trace The error trace.
+	 * @return string The sanitized trace.
+	 */
+	private function sanitize_trace( $trace ) {
```

Comment:
Optionally, could specify the return type here:

```suggestion
	private function sanitize_trace( $trace ): string {
```